App.js

```diff
@@ -43,20 +71,40 @@ export default function App() {
       console.error('Error adding document: ', e);
     });
 
+  let [fontsLoaded] = useFonts({
+    OpenSans_300Light,
+    OpenSans_400Regular,
+    OpenSans_600SemiBold,
+    OpenSans_700Bold,
+    Montserrat_300Light,
+    Montserrat_400Regular,
+    Montserrat_500Medium,
+    Montserrat_600SemiBold,
+    Montserrat_700Bold,
+  });
   const styles = StyleSheet.create({
     container: {
       flex: 1,
-      backgroundColor: '#fff',
+      backgroundColor: 'rgba(243,225,210,0.4)',
       alignItems: 'center',
       justifyContent: 'center',
     },
   });
-  return (
-    <BiteShareContext.Provider value={{ state, dispatch }}>
-      <ThemeProvider theme={theme}>
-        <DummyComponent />
-      </ThemeProvider>
-
-    </BiteShareContext.Provider>
-  );
+  if (!fontsLoaded) {
+    // console.log('L87, fonts not found');
+    return (
+      <AppLoading />
+    );
+  } else {
+    return (
+      <BiteShareContext.Provider value={{ state, dispatch }}>
+        <ThemeProvider theme={theme}>
+          <HostQR />
+          <DummyComponent />
+        </ThemeProvider>
+
+      </BiteShareContext.Provider>
+    );
+  }
+
```

Comment:
not blocking but maybe a turnary would be cleaner? (not sure whether it works though lol)
```suggestion
  return (
     <BiteShareContext.Provider value={{ state, dispatch }}>
       { fontsLoaded
         ? <AppLoading />
         : <ThemeProvider theme={theme}>
             <HostQR />
             <DummyComponent />
         </ThemeProvider>
       }
     </BiteShareContext.Provider>
  )

```

Reply:
yes. It's work. I just changed the logic. Thanks Tingjun.